### DISCUSSION TITLE
Applying Scale

### DISCUSSION THREAD

**Discussion by drinkspiller** [fabricjs/fabric.js#10159] [Q&A]
I am trying to make a simple demo of a FixedLayout Group containing a rect. The rect is not selectable.

When the group is scaled, the goal is:
1. apply the scale so that the Groups new width and height are the previous scaled width and height and scaleX and scaleY are set to one.
2. Keep the upper left of the child rect relative to the upper left of its parent group

Setting aside #2 for the moment, just applying the scale on resize results in the right-middle resize control not responding and the left-middle resize control moving the block instead of resizing it:

![2024-09-16_06-33-35 (1)](https://github.com/user-attachments/assets/14c0f280-9510-465e-953e-4b51c26440bc)

Code:
https://codepen.io/drinkspiller/pen/GRbaPyX?editors=0010

It _feels_ like buggy behavior, but I suspect I might be missing something. I appreciate any guidance, thank you!

**Comment by asturur** [comment]
Di you look into customizing controls to not scale but just change width/height property of the group? 
There is no reason to run a scale action if you do not want to scale.

**Comment by drinkspiller** [comment]
Thank you for the reply, much appreciated. I certainly can look into that but shouldn't this work?

**Reply by asturur** [reply]
Yes it should work but with an animated gif is hard to undertand why is not working.

**Reply by drinkspiller** [reply]
(there is a link to the code that produced the GIF in the original post)

**Reply by asturur** [reply]
Ok i didn't see the link the first time, my bad.


for #1 You used getScaledHeight() for both width and height. Unless that is a way to keep the group square, but in this case is blocking you from growing by width.
<img width="360" alt="image" src="https://github.com/user-attachments/assets/07a2c03e-7001-4830-8059-d6689de36a03">

for #2 I think what you are looking for is building your own layout manager or custom actions. I don't really have a suggestion on what is more appropriate or easy to do because i never went deep into customizing one layout manager

**Comment by drinkspiller** [comment]
Doh — indeed that was the issue with #1 — thanks for catching that mistake!

Re: #2 — understood, I'll explore a custom LayoutManager/LayoutStrategy. Cheers!
